addons/mail/models/discuss/discuss_channel.py

```diff
@@ -647,6 +647,13 @@ def _notify_thread(self, message, msg_vals=False, **kwargs):
             ),
             (self, "discuss.channel/new_message", payload),
         ]
+        if "partner_ids" in msg_vals and not self.env.user._is_public():
+            member_partner_ids = self.env['discuss.channel.member'].with_context(active_test=False).search([
+                ('partner_id', '!=', self.env.user.partner_id.id),
+                ('channel_id', '=', self.id)
+            ]).mapped('partner_id.id')
+            for p in self.env["res.partner"].browse([pid for pid in msg_vals["partner_ids"] if pid not in member_partner_ids]):
+                bus_notifications.append((p, "mail.message/inbox", Store(message.with_user(p), msg_vals=msg_vals).get_result()))
```

Comment:
`_notify_thread` is already doing `_notify_thread_by_inbox` which sends the `mail.message/inbox` notification. Why this code? If needed for good working, this looks more something to improve in `_notify_thread_by_inbox` rather than custom code here.